pkg/object/httpserver/mux.go

```diff
@@ -431,3 +467,34 @@ func (mi *muxInstance) exportPrometheusMetrics(stat *httpstat.Metric, backend st
 	mi.metrics.RequestSizeBytesPercentage.With(labels).Observe(float64(stat.ReqSize))
 	mi.metrics.ResponseSizeBytesPercentage.With(labels).Observe(float64(stat.RespSize))
 }
+
+func newAccessLogFormatter(format string) *accessLogFormatter {
+	if format == "" {
+		format = defaultAccessLogFormat
+	}
+	escapeReg := regexp.MustCompile(`(\{|\})`)
+	expr := escapeReg.ReplaceAllString(format, "{{`$1`}}")
+	varReg := regexp.MustCompile(`\$([a-zA-z]*)`)
+	expr = varReg.ReplaceAllString(expr, "{{.$1}}")
+	tpl := template.Must(template.New("").Parse(expr))
+	return &accessLogFormatter{template: tpl}
+}
+
+func (formatter *accessLogFormatter) format(log *accessLog) string {
+	var buf bytes.Buffer
+	if err := formatter.template.Execute(&buf, log); err != nil {
+		logger.Errorf("format access log failed: %v", err)
+	}
+	return buf.String()
+}
+
+func printHeader(header protocols.Header) string {
```

Comment:
we know the header is an HTTP one, so please use `http.Header` as the argument type.